packages/cpp-debug/src/browser/cpp-memory-widget.tsx

```diff
@@ -308,37 +373,100 @@ export class MemoryView extends ReactWidget {
 
                 // The ASCII view is always in strictly increasing address order.
                 for (const byte of groupBytes) {
-                    asciiStr += isprint(byte) ? String.fromCharCode(byte) : '.';
+                    asciiStr += typeof byte === 'undefined' ?
+                        ' ' : isprint(byte) ? String.fromCharCode(byte) : '.' ;
                 }
             }
 
-            rows.push([addressStr, rowBytesStr, asciiStr])
+            rows.push([rowAddr, addressStr, rowBytesStr, asciiStr])
         }
 
         return <React.Fragment>
             {
                 rows.map((row, index) =>
-                    <tr className='t-mv-view-row' key={index}>
-                        <td className='t-mv-view-address'>{row[0]}</td>
-                        <td className='t-mv-view-data'>{row[1]}</td>
-                        <td className='t-mv-view-code'>{row[2]}</td>
+                    <tr className={ 't-mv-view-row' + (
+                        // Add a marker to help visual navigation when scrolling
+                        row[0] % (this.bytesPerRow * 8) < this.bytesPerRow ?
+                            ' t-mv-view-row-highlight' : '')
+                    } key={index}>
+                        <td className='t-mv-view-address'>{row[1]}</td>
+                        <td className='t-mv-view-data'>{row[2]}</td>
+                        <td className='t-mv-view-code'>{row[3]}</td>
                     </tr>
                 )
             }
         </React.Fragment>;
     }
 
+    protected onWheelMemoryView = (event: React.WheelEvent) => {
+        if (this.memoryReadResult) {
+            const scroll = event.currentTarget.scrollTop;
+            const canScrollDown = scroll >= (event.currentTarget as any).scrollTopMax;
+            const canScrollUp = scroll <= 0;
+
+            const { deltaY } = event;
+            const step = this.bytesPerRow * 2;
+
+            if (canScrollUp && deltaY < 0) { // wheel up: go up
+                this.locationOffset -= step;
+            } else if (canScrollDown && deltaY > 0) { // wheel down: go down
+                this.locationOffset += step;
+            } else {
+                return;
+            }
+            this.update();
+            this.updateMemoryView();
+        }
+    };
+
+    protected onKeyboardMemoryView = (event: React.KeyboardEvent) => {
+        if (this.memoryReadResult) {
+            const scroll = event.currentTarget.scrollTop;
+            const canScrollDown = scroll >= (event.currentTarget as any).scrollTopMax
```

Comment:
similar to above:
```ts
const canScrollDown = scroll >= 0
```

Reply:
I don't think it will be enough, given that I test `scroll <= 0` in order to scroll up, I really expect the scrolling to get up to some `x != 0` representing the other end of the elevator.

Did you see the values?